mne/preprocessing/nirs/_optical_density.py

```diff
@@ -32,20 +34,26 @@ def optical_density(raw):
         raw.info, np.unique(_channel_frequencies(raw.info, nominal=True)))
 
     picks = _picks_to_idx(raw.info, 'fnirs_cw_amplitude')
-    data_means = np.mean(raw.get_data(), axis=1)
 
     # The devices measure light intensity. Negative light intensities should
     # not occur. If they do it is likely due to hardware or movement issues.
     # Set all negative values to abs(x), this also has the benefit of ensuring
     # that the means are all greater than zero for the division below.
     if np.any(raw._data[picks] <= 0):
         warn("Negative intensities encountered. Setting to abs(x)")
-        raw._data[picks] = np.abs(raw._data[picks])
-
-    for ii in picks:
-        raw._data[ii] /= data_means[ii]
-        np.log(raw._data[ii], out=raw._data[ii])
-        raw._data[ii] *= -1
-        raw.info['chs'][ii]['coil_type'] = FIFF.FIFFV_COIL_FNIRS_OD
+        min_ = np.inf
+        for pi in picks:
+            np.abs(raw._data[pi], out=raw._data[pi])
+            min_ = min(min_, raw._data[pi].min() or min_)
+        # avoid == 0
+        for pi in picks:
+            np.maximum(raw._data[pi], min_, out=raw._data[pi])
```

Comment:
Why do you set it to _min and not something like eps?

Reply:
`eps` is relative to `1.` and is around 1e-12. Imagine you had data in the femto/1e-15 or smaller range, your `np.finfo(float).eps` would actually be a "large" value relative to this scale.

Reply:
Thanks @larsoner, turns out I didnt understand eps properly. I wonder how many bugs I have introduced over the years!